### PR TITLE
Implement support for VCF 452 APIs

### PR DESCRIPTION
**Pull request by dimitarproynov**
The update is required as new APIs for rotation of certificate from external CA have been included in the 452 release, that the VCF provider needs to make use of.

Reverted manually changes to ResourcePoolSpec and SDDCNSXTSpec, changed in previous commits.

In order to have a good experience with our community, we recommend that you read the [contributing guidelines](https://github.com/vmware/vcf-sdk-go/blob/main/CONTRIBUTING_DCO.md) for making a pull request.

**Summary of Pull Request**

Implement support for VCF 452 APIs
**Type of Pull Request**

- [ ] This is a bug fix.
- [X] This is an enhancement or feature.
- [ ] This is a code style/formatting update.
- [ ] This is a documentation update.
- [ ] This is a refactoring update.
- [ ] This is a chore update
- [ ] This is something else.
      Please describe:

**Related to Existing Issues**

Issue Number: N/A

**Test and Documentation Coverage**

Testing done:
go mod tidy
go build

For bug fixes or features:

- [X] Tests have been completed.
- [ ] Documentation has been added/updated.

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [X] No, there are no breaking changes.

### CODE REVIEW

File: swagger.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "swagger" : "2.0",
   "info" : {
     "description" : "VMware Cloud Foundation is an integrated software stack that bundles the VMware virtualization and management components like compute virtualization (vSphere), storage virtualization (vSAN), network virtualization (NSX), and cloud management (vRealize Suite). VCF consists of Cloud Builder - that automates the bringup of the management infrastructure and SDDC Manager - that automates the workload domain management, lifecycle and operations. This API reference guide covers APIs of both Cloud Builder and SDDC Manager",
-    "version" : "4.5.0",
+    "version" : "4.5.2-vcf4520RELEASE",
```
Comment (raj-chelur) [inline swagger.json:5]: This version string looks odd. Did it come from the release branch?

**Review by raj-chelur — APPROVED** [review]

**Review by vasilsatanasov — APPROVED** [review]
LGTM!

**Review by tenthirtyam — DISMISSED** [review]
LGTM! 🏗️

File: swagger.json (hunk 1 above)
Comment (dimitarproynov) [reply swagger.json:5]: It came from the recommended build (the same for some mouths now) from vcf4520. Maybe I should leave "4.5.2" only?

**Comment by raj-chelur** [comment]
Yes. Please change it to 4.5.2.

**Review by vasilsatanasov — APPROVED** [review]
LGTM!

**Pull request merged**